os/drivers/input/ist415.c

```diff
@@ -568,6 +573,11 @@ static int ist415_resume_device(struct touchscreen_s *upper)
 		ASSERT(get_errno() == EINTR);
 	}
 
+	if (dev->forcedoff) {
```

Comment:
How about checking this at start of the function? Then, you don't need to take the sem.